src/notebooks/controllers/controllerRegistration.ts

```diff
@@ -179,6 +197,75 @@ export class ControllerRegistration implements IControllerRegistration {
         const id = this.getControllerId(metadata, notebookType);
         return this.registeredControllers.get(id);
     }
+    @swallowExceptions()
+    private async swapKernelSpecConnectionControllerWithLiveController(kernel: IKernel) {
+        if (kernel.kernelConnectionMetadata.kind === 'startUsingRemoteKernelSpec') {
+            if (!kernel.session?.kernel?.id) {
+                return;
+            }
+            // Check if we have a controller registered for this live kernel session.
+            // If not, then create one.
+            // Next, then swap the metadata & controller in the kernel.
+            const liveKernelConnection: LiveRemoteKernelConnectionMetadata = {
+                kind: 'connectToLiveRemoteKernel',
+                baseUrl: kernel.kernelConnectionMetadata.baseUrl,
+                id: kernel.session.kernel.id,
+                kernelModel: {
+                    ...kernel.kernelConnectionMetadata.kernelSpec,
+                    model: {
+                        id: kernel.session.kernel.id,
+                        name: kernel.session.kernel.name,
+                        path: path.basename(kernel.uri.path),
+                        type: 'notebook',
+                        kernel: kernel.session.kernel.model
+                    },
+                    notebook: {
+                        path: path.basename(kernel.uri.path)
+                    },
+                    name: kernel.session.kernel.name,
+                    numberOfConnections: 1,
+                    lastActivityTime: new Date(Date.now()),
+                    id: kernel.session.kernel.id
+                },
+                serverId: kernel.kernelConnectionMetadata.serverId,
+                interpreter: kernel.kernelConnectionMetadata.interpreter
+            };
+            if (!this.registeredControllers.has(this.getControllerId(liveKernelConnection, 'jupyter-notebook'))) {
+                this.add(liveKernelConnection, [JupyterNotebookView, InteractiveWindowView]);
+            }
+            const liveController = Array.from(this.registeredControllers.values()).find(
```

Comment:
Array.from shouldn't be necessary here.

Reply:
Yes it is required, i'm iterating through the values in a Map (not an array)